macosx/Controller.mm

```diff
@@ -644,6 +646,33 @@ - (void)awakeFromNib
     }
 
     //load previous transfers
+    tr_torrent** torrentSet;
+    tr_ctor* ctor = tr_ctorNew(self.fLib);
+
+    //start all torrents paused then check state in history
+    tr_ctorSetPaused(ctor, TR_FORCE, true);
+
+    torrentSet = tr_sessionLoadTorrents(self.fLib, ctor, nullptr);
+    tr_ctorFree(ctor);
+    tr_free(torrentSet);
+        
+    auto* session = static_cast<tr_session*>(self.fLib);
+    auto torrents = tr_sessionGetTorrents(session);
+    for (struct tr_torrent* tor : torrents)
```

Comment:
This new function isn't necessary because`tr_sessionLoadTorrents()` returns the same thing already.

Would recommend removing `tr_sessionGetTorrents()` from `libtransmission/transmission.h` and then using the array returned from `tr_sessionLoadTorrents()` instead, e.g.

```
auto* const session = self.fLib;

// load previous torrents
tr_ctor* ctor = tr_ctorNew(session);
tr_ctorSetPaused(ctor, TR_FORCE, true); // paused by default; unpause below after checking state history
int n_torrents = 0;
tr_torrent** loaded_torrents = tr_sessionLoadTorrents(session, ctor, &n_torrents);
tr_ctorFree(ctor);

// process the loaded torrents
for (int i = 0; i < n_torrents; ++i)
{
    auto* const tor = loaded_torrents[i];
    ....
}

tr_free(loaded_torrents);
```

Reply:
was getting `TR_ASSERT(tr_isTorrent(tor));`
at this line:
`auto* const tor = loaded_torrents[I];`

which was resolved like so:
`struct tr_torrent* tor = loaded_torrents[I];`

interestingly enough, after using the second snippet once, then reverting to your suggestion - no more crashes...
Ive left in the specific declaration just to be safe...